src/utils/hooks/useAlerts/utils/utils.ts

```diff
@@ -0,0 +1,87 @@
+import { murmur3 } from 'murmurhash-js';
+
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import {
+  Alert,
+  AlertStates,
+  PrometheusRule,
+  Rule,
+  RuleStates,
+  Silence,
+  SilenceStates,
+} from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Group,
+  MONITORING_SALT,
+  PrometheusRulesResponse,
+} from '@virtualmachines/details/tabs/overview/utils/utils';
+
+export const addAlertIdToRule = (group: Group, rule: PrometheusRule): Rule => {
+  const key = [
+    group?.file,
+    group?.name,
+    rule?.name,
+    rule?.duration,
+    rule?.query,
+    ...Object?.entries(rule?.labels).map(([k, v]) => `${v}=${k}`),
+  ].join(',');
+  return { ...rule, id: String(murmur3(key, MONITORING_SALT)) };
+};
+
+export const getAlertsAndRules = (
+  data: PrometheusRulesResponse['data'],
+): { alerts: Alert[]; rules: Rule[] } => {
+  // Flatten the rules data to make it easier to work with, discard non-alerting rules since those
+  // are the only ones we will be using and add a unique ID to each rule.
+  const groups = data?.groups as PrometheusRulesResponse['data']['groups'];
```

Comment:
there is a type for group at src/views/virtualmachines/details/tabs/overview/utils/utils.tsx

Reply:
Fixed!